app/src/main/java/com/example/weathertimeapp/activities/MainActivity.kt

```diff
@@ -15,8 +15,9 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
         presenter = WeatherTimePresenter(WeatherTimeModel(readCitiesJSONFile()), WeatherTimeView(this))
```

Comment:
change this to use the contract pls :D
presenter: WeatherTimeContract.Presenter = WeatherTimePresenter(WeatherTimeModel(readCitiesJSONFile()), WeatherTimeView(this))

Reply:
it is done with a late init at the beginning of the class